server/xpub-model/entities/manuscript/index.test.js

```diff
@@ -178,28 +263,45 @@ describe('Manuscript', () => {
       ).rejects.toThrow('Manuscript not found'))
   })
 
+  describe('refresh()', () => {
+    it('refreshes manuscript', async () => {
+      await testOverwritingWithOldData(userId, true)
```

Comment:
I'm not a fan of having tests like this. It makes it difficult to see what's being tested and what's being asserted on. I use the tests a lot like documentation, and extracting tests out like this makes it difficult to see how to call a function and what the behaviour of it is.

I'm happy extracting mocks and setup code to separate functions, but the invocation of the unit under test and the assertions should be more visible for each test.

If we really want to make our own DSL for the tests to make the more DRY, we'll want to make sure the tests are still readable.

Reply:
Can you point out why you think this is not readable?

Reply:
The given/when/then (arrange/act/assert) phases are not visible, in fact this is correlated to an `if` inside the function being called here. It could be split into pieces which would also avoid the `if`. Very pseudo-code like:
```
v1 = givenManuscriptVersion(1)
v2 = modifyManuscriptVersion(v1)
v3 = modifyManuscriptVersion(v1)
save(v3)
refresh(v2)
assertNoChanges(v2) # or any assert...(...)
```

Reply:
That makes sense ... I guess what you're asking is not to use functions in the (arrange>act>assert) categories rather than in its entirety.